### PR TITLE
RFC: replace deprecated `matplotlib.pyplot.plot_date` function in docs and tests

### LINKED ISSUE #17317: pyplot.plot_date is deprecated
Discovered while working on #17316:

`matplotlib.pyplot.plot_date` and `matplotlib.axes.Axes.plot_date` are deprecated, as per [its documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.plot_date.html#matplotlib.pyplot.plot_date), since matplotlib 3.9, and set to be removed in 3.11 (3.10 is currently in rc, so no need to panic yet).
We're using them only in a couple doctest, but:
1) the replacement (`plot`) doesn't produce identical results out of the box.
For instance (adapted from one of our doctests):

```python
import numpy as np
from matplotlib import pyplot as plt

from astropy import units as u
from astropy.time import Time
from astropy.coordinates import EarthLocation, AltAz, GCRS

time = Time('J2010') + np.linspace(-1,1,1000)*u.min
location = EarthLocation(lon=0*u.deg, lat=45*u.deg)
aa = AltAz(alt=[45]*1000*u.deg, az=90*u.deg, distance=100*u.au,
            radial_velocity=[10]*1000*u.km/u.s,
            location=location, obstime=time)
gcrs = aa.transform_to(GCRS(obstime=time))

fig, ax = plt.subplots()
ax.plot_date(time.plot_date, gcrs.radial_velocity.to(u.km/u.s))
ax.set_ylabel('RV [km/s]')

sfile = "/tmp/test_plot_date.png"
fig.savefig(sfile)
print(f"saved to {sfile}")


fig, ax = plt.subplots()
ax.plot(time.plot_date, gcrs.radial_velocity.to(u.km/u.s))
ax.set_ylabel('RV [km/s]')
sfile = "/tmp/test_plot.png"
fig.savefig(sfile)
print(f"saved to {sfile}")
```
produces:
test_plot_date
test_plot

at this stage I don't know if this is expected, a bug on our side, or a bug in matplotlib. Further inquiry is needed.

2) `TimePlotDate` is documented as `Matplotlib `~matplotlib.pyplot.plot_date` input` so it'll need to be changed

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Fixes #17317

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/coordinates/velocities.rst
Comment (eerovaher) [inline docs/coordinates/velocities.rst:424]: You are editing the same lines in #17316. Which order should the pull requests be reviewed in?

Comment (neutrinoceros) [reply docs/coordinates/velocities.rst:424]: Order doesn't matter, I'll need to handle a conflict in all cases.

File: astropy/coordinates/tests/test_finite_difference_velocities.py
Comment (eerovaher) [inline astropy/coordinates/tests/test_finite_difference_velocities.py:299]: I wonder what was the last time anyone called this function. Maybe we should just delete it.

File: docs/coordinates/velocities.rst
Comment (eerovaher) [inline docs/coordinates/velocities.rst:424]:
```suggestion
    ax.scatter(time.plot_date, gcrs.radial_velocity.to(u.km / u.s))
    ax.set_ylabel("RV [km/s]")
```

File: docs/coordinates/velocities.rst
Comment (eerovaher) [inline docs/coordinates/velocities.rst:446]:
```suggestion
    ax.scatter(time.plot_date, gcrs.radial_velocity.to(u.km / u.s))
    ax.set_ylabel("RV [km/s]")
```

File: astropy/coordinates/tests/test_finite_difference_velocities.py
Comment (eerovaher) [reply astropy/coordinates/tests/test_finite_difference_velocities.py:299]: `diff_info_plot()`

Comment (neutrinoceros) [reply astropy/coordinates/tests/test_finite_difference_velocities.py:299]: oh yes. Maybe as a follow up PR then, just in case anyone complains about its removal after the fact so we can easily revert it ?

Comment (eerovaher) [reply astropy/coordinates/tests/test_finite_difference_velocities.py:299]: `diff_info_plot()` is defined in our tests, so we can be sure normal users are not relying on this. So it really boils down to whether or not @adrn, @mhvk or @StuartLittlefair are interested in keeping it.

Comment (mhvk) [reply astropy/coordinates/tests/test_finite_difference_velocities.py:299]: I've never used it, even though I do sometimes use tests interactively in debugging. I think the chance of knowing of its existence is far too small, so agree with @eerovaher that we should just remove it. But just in case, ping @eteq, who I think introduced it.

File: astropy/time/formats.py
Comment (mhvk) [inline astropy/time/formats.py:958]: Somewhat unrelated to the PR, but why not just `fig.show()` - seems easier and thus better for an interactive example (May need to add `# doctest: +SKIP` if this causes problems.)

File: docs/time/index.rst
Comment (mhvk) [inline docs/time/index.rst:1017]: This misses a `ax.xaxis.axis_date()`, I think.

Though I think that also here, we should perhaps just go for using `t.datetime` to automatically gets dates...

File: docs/coordinates/velocities.rst
Comment (mhvk) [inline docs/coordinates/velocities.rst:420]: How about we replace this with the new way of doing dates, which is to pass in a `datetime`, i.e.,
```
plt.plot(time.datetime, gcrs.radial_velocity.to(u.km/u.s))
plt.ylabel('RV [km/s]')
```

File: docs/coordinates/velocities.rst
Comment (mhvk) [inline docs/coordinates/velocities.rst:439]: Same here

**Review by mhvk — COMMENTED** [review]
This looks good! It is a bit of a pity that the examples all become longer, but we cannot help that.

That said, if one uses `t.datetime` instead of `t.plot_date`, this gets done automatically.

I also wonder if we cannot support this via `astropy.visualization.time_support`, but that would be for another issue/PR...

I also wondered if we should use `t.datetime` instead, since then `matplotlib` deals with dates automatically. That said, I'm not sure how leap seconds are dealt with in either case, and am running out of time to test. Maybe good to ping @taldcroft

**Comment by neutrinoceros** [comment]
rebased to fix a conflict from #17409

File: docs/coordinates/velocities.rst
Comment (eerovaher) [inline docs/coordinates/velocities.rst:444]: This creates two separate plots where there is a single combined plot in `main`.

Comment (neutrinoceros) [reply docs/coordinates/velocities.rst]: thanks for catching this ! fixed

**Review by eerovaher — APPROVED** [review]
It is not obvious to me if the `coordinates` plots being edited here are needed at all, but it is not obvious to me that they are not needed either. In this pull request it is therefore best to just replace deprecated code and not to worry too much about the necessity of the plots.

I am approving the changes to `coordinates`, but someone else still needs to approve for `time`.

File: docs/time/index.rst
Comment (pllim) [inline docs/time/index.rst:1016]: I don't understand why everything is skipped and it is not even a plot directive, so no nice plot is being added to the doc. Maybe a `time` maintainer should review.

**Review by taldcroft — APPROVED** [review]
This looks good for `time` and in general 👍 on using `.datetime` for plotting for these examples.

File: docs/time/index.rst
Comment (pllim) [reply docs/time/index.rst:1016]: Okay, thanks for your feedback. I opened https://github.com/astropy/astropy/issues/17454 as follow-up issue.

**Pull request merged**
